privacyidea/lib/smsprovider/HttpSMSProvider.py

```diff
@@ -64,6 +65,9 @@ def submit_message(self, phone, message):
         log.debug("submitting message {0!r} to {1!s}".format(message, phone))
         parameter = {}
         if self.smsgateway:
+            regexp = self.smsgateway.option_dict.get("REGEXP", "")
+            if regexp != "":
+                phone = re.sub(regexp, "", phone)
```

Comment:
We can add the phone mangling lines before the if-statement in line 67.
Also remove lines 87,88,89.
In both cases the phone number would be stripped, so do it before the if-statement.